lib/find/find-references.js

```diff
@@ -248,11 +248,20 @@ function urlToPath(value, config, type) {
 
     value = url.pathname.slice(config.urlConfig.prefix.length)
 
+    // Drop the first path segment (the branch) if the prefix is
+    // branch-less (matches every branch). Make an educated guess
+    // whether the prefix is branch-less: Currently every
+    // hosted-git-info browsetemplate/pathtemplate has <= 2 segments
+    // (`${user}/$project}`), plus e.g. "blob" and surrounding slashes
+    // -> 5.
+    //
     // Things get interesting here: branches: `foo/bar/baz` could be `baz` on
     // the `foo/bar` branch, or, `baz` in the `bar` directory on the `foo`
     // branch.
     // Currently, we’re ignoring this and just not supporting branches.
-    value = value.split(slash).slice(1).join(slash)
+    if (config.urlConfig.prefix.split('/').length <= 5) {
+      value = value.split(slash).slice(1).join(slash)
+    }
```

Comment:
This is way too magic? It doesn’t seem “educated”? Who says branches are 5+ characters?

Reply:
What I was wanting to get at wasn't the characters:
```js
'/DefinitelyTyped/DefinitelyTyped/blob/'.split('/')`
// -> [ '', 'DefinitelyTyped', 'DefinitelyTyped', 'blob', '' ]
```

so `.length <= 5`. Currently this is true of every hosted-git-info `browsetemplate`/`pathtemplate`: https://github.com/npm/hosted-git-info/blob/702b7df467f2fd4bf46395134c62534bb966ca7b/git-host-info.js#L8